### ISSUE TITLE
Warning if epoch non-training overhead take significant time compared to training

### ISSUE THREAD

**Issue by turian** [PyTorchLightning/pytorch-lightning#10660]
## 🚀 Feature

If training epoch overhead (e.g. logging) tasks significant time compared to gradient update steps, a warning should be logged with suggestions on epoch parameters that should be changed to reduce training overhead.

### Motivation

This is related to https://github.com/PyTorchLightning/pytorch-lightning/pull/2367

### Pitch

Sometimes training is very fast, e.g. all examples are cached in the GPU.

However, ptl can be much slower than pure torch in this case, because there is additional ptl per-epoch overhead. Logging is one, and perhaps others that a casual user might not be aware of.

Explaining to the user that a good speedup can occur by tweaking other Trainer variables would be very helpful for speed.

### Alternatives

Each user with fast training epochs must carefully understand all the per-epoch ptl operations that incur overhead.

cc @borda @rohitgr7

**Comment by cowwoc** [comment]
@awaelchli You asked for a discussion, @turian created it, and it went nowhere :)

I agree that it would be helpful to provide a guide for increasing GPU usage. I have never been able to get GPU usage over 3%. From the sound of it, @turian is having a similar problem. I assume other users are as well.

Is it possible to work with @turian to figure out what is going wrong in his case and build up a guide to point users in the right direction of debugging this kind of problem?

**Comment by cowwoc** [comment]
Slight correction to the above post... @awaelchli asked for a Github discussion thread, not a new PR. @turian did you ever create one?

Wherever you guys are having this discussion, I would love to be copied on it.

**Comment by turian** [comment]
@cowwoc https://github.com/Lightning-AI/lightning/discussions/16519
